File: client/src/main/java/io/xol/chunkstories/input/lwjgl3/Lwjgl3KeyBindCompound.java
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package io.xol.chunkstories.input.lwjgl3;

import io.xol.chunkstories.api.input.KeyboardKeyInput;

public class Lwjgl3KeyBindCompound extends Lwjgl3Input implements KeyboardKeyInput {

	final String defaultKeysNames;
	int[] glfwKeys;
	
	public Lwjgl3KeyBindCompound(Lwjgl3ClientInputsManager im, String name, String defaultKeysNames) {
		super(im, name);
		
		this.defaultKeysNames = defaultKeysNames;
		
		reload();
	}

	@Override
	public boolean isPressed() {
		return false;
	}

	@Override
	public void reload() {
		//TODO TODO TODO
		String keyNamesString = defaultKeysNames; //Client.getInstance().getConfig().getString("bind.glfw.compound."+name, defaultKeysNames);
		String keyNames[] = keyNamesString.split("\\+");
		
		glfwKeys = new int[keyNames.length];
		for(int i = 0; i < keyNames.length; i++) {
			String keyName = keyNames[i];
			
			int glfwKey = GLFWKeyIndexHelper.getGlfwKeyByName(keyName);
			glfwKeys[i] = glfwKey;
			//System.out.println(keyName+":"+glfwKey);
		}
		
		this.im.logger().debug("Initialized keyBindCompound "+name+" for "+glfwKeys.length+" keys.");
	}

}

File: client/src/main/java/xyz/chunkstories/input/lwjgl3/Lwjgl3KeyBindCompound.kt
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.input.lwjgl3

import org.lwjgl.glfw.GLFW
import xyz.chunkstories.api.input.KeyboardKeyInput

class Lwjgl3KeyBindCompound(inputsManager: GLFWInputManager, name: String, internal val defaultKeysNames: String) : Lwjgl3Input(inputsManager, name), KeyboardKeyInput {
    internal var glfwKeys: IntArray

    init {
        val keyNamesString = defaultKeysNames
        val keyNames = keyNamesString.split("\\+".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

        glfwKeys = IntArray(keyNames.size)
        for (i in keyNames.indices) {
            val keyName = keyNames[i]

            val glfwKey = GLFWKeyIndexHelper.getGlfwKeyByName(keyName)
            glfwKeys[i] = glfwKey ?: GLFW.GLFW_KEY_UNKNOWN
            // System.out.println(keyName+":"+glfwKey);
        }

        this.inputsManager.logger().debug("Initialized keyBindCompound " + name + " for " + glfwKeys.size + " keys.")
    }

    override val isPressed: Boolean
        get() {
            //TODO
            return false
        }

}
